api/v1alpha3/virtualmachineimage_types.go

```diff
@@ -81,6 +82,40 @@ type VirtualMachineImageProductInfo struct {
 	FullVersion string `json:"fullVersion,omitempty"`
 }
 
+// VirtualMachineImageDiskInfo describes disk information for an image taken from DiskSection
```

Comment:
Suggest perhaps:

```golang
VirtualMachineImageDiskInfo describes information about any disks associated with
this image.
```

I'm not sure we need to leak where the information is derived.

Reply:
I was working with the requirement `should report the disk layout, including size, capacity, provisioning type, etc.`. The only one-to-one correspondence, and the ones that make sense in my mind, are capacity and size. But with the presence of the other explicitly named items and the presence of the `etc.`, I erred on the side of populating the full disk info as it appears in the envelope.

Reply:
Updated with suggestion